### PR TITLE
update java version and add room dependencies

### PR DESCRIPTION
**Pull request by sabrina-li**
gradle change to update languageLevel to 1.8 to support lambda expression, and adding room dependencies

### CODE REVIEW

File: java/app/build.gradle [hunk 1]
```diff
@@ -28,7 +32,12 @@ dependencies {
     implementation 'androidx.navigation:navigation-fragment:2.0.0'
     implementation 'androidx.navigation:navigation-ui:2.0.0'
     implementation 'androidx.lifecycle:lifecycle-extensions:2.0.0'
+    def room_version = "2.2.4"
```
Comment (van-fs) [inline java/app/build.gradle]: Some observations:
- the `def` kinda breaks up the dependencies, is it a better practice in gradle to group/move these to the start of the dependencies block?
- we have a `room_version` that defines a version but all other versions are hardcoded; so there's reusability but inconsistency with how the dependencies are defined
- this is using double quotes versus the single quotes that existed

Comment (sabrina-li) [reply java/app/build.gradle]: great point! I moved all the versions to ext for easier management. 

a couple good reads during research on this topic:
https://developer.android.com/studio/build/gradle-tips.html#configure-project-wide-properties
https://handstandsam.com/2018/02/11/kotlin-buildsrc-for-better-gradle-dependency-management/

File: java/app/build.gradle [hunk 2]
```diff
@@ -17,18 +17,34 @@ android {
...
+    nav_version = "2.2.1"
+    lifecycle_version = "2.2.0"
+    constraintlayout_version = "1.1.3"
+    appcompat_version = "1.1.0"
+    junit_version = "1.1.1"
+    espresso_version = "3.2.0"
 }
 
 dependencies {
     implementation fileTree(dir: 'libs', include: ['*.jar'])
```
Comment (patrick-fs) [inline java/app/build.gradle]: NIT: double vs single-quote consistency

File: java/app/build.gradle [hunk 3]
```diff
@@ -17,18 +17,34 @@ android {
...
-    implementation 'androidx.lifecycle:lifecycle-extensions:2.0.0'
+    implementation "androidx.appcompat:appcompat:$appcompat_version"
+    implementation "androidx.constraintlayout:constraintlayout:$constraintlayout_version"
+    implementation "androidx.navigation:navigation-fragment:$nav_version"
+    implementation "androidx.navigation:navigation-ui:$nav_version"
+    implementation "androidx.lifecycle:lifecycle-extensions:$lifecycle_version"
+    implementation "androidx.room:room-runtime:$room_version"
+    annotationProcessor "androidx.room:room-compiler:$room_version"
+
     testImplementation 'junit:junit:4.12'
```
Comment (patrick-fs) [inline java/app/build.gradle]: NIT: double vs single-quote consistency

Comment (sabrina-li) [reply java/app/build.gradle]: Looking at the Android doc, seems like a norm for the versions value to have double quote, while dir inside fileTree to have single quote for arbitrary string. While double quotes are required for using version variable.

**Pull request merged**